### ISSUE TITLE
Base Movement seems bugged?

### ISSUE THREAD

**Issue by Pangolinx** [jaszhix/NoMansConnect#138]
Not sure if this is a known issues or not; I'm playing in the current version of NMS (Beyond, patched, not experimental).  I'm just starting to use NoMansConnect.

I created a test base and then tried to move it to another location by overwriting a base there.  The old base appears to have been correctly deleted, but the base I attempted to move showed up in scattered/rotated pieces; i.e., every part was there, but they were rotated around in different ways and also relocated relative to each other, including some overlapping.  Looks a lot like a Salvador Dali painting.

Is this a known issue?  I know base relocation is not the primary purpose of NMC but it's a feature I was super excited about, since I haven't seen any other mod/program that does this, and it lets me build a base without having to worry if it's on the "perfect" planet or not.

Very grateful for any help, and thanks for developing this cool utility!
![20191002121240_1](https://user-images.githubusercontent.com/56085554/66062191-f12be780-e50e-11e9-9182-3bf5696c6b8a.jpg)

**Comment by jaszhix** [comment]
This is a known issue, but will keep this open for visibility and in case someone wants to look at a fix for it. It used to work but the NEXT update broke it.

**Comment by Pangolinx** [comment]
> This is a known issue, but will keep this open for visibility and in case someone wants to look at a fix for it. It used to work but the NEXT update broke it.

Thanks, jaszhix!  I saw a note in one of the posts for NMC noting that monkeyman192 (at github) had the credit for implementing the base moving formula.  I am new to github and found his profile here but couldn't figure out how to message him to see if he'd have any interest in taking a look at the problem.  Do you know if there's a way to get in touch with him without being too intrusive?

**Comment by jaszhix** [comment]
> Do you know if there's a way to get in touch with him without being too intrusive?

We can try pinging him here. I did reach out on Discord a while ago, but he seemed unsure at the time. But can't hurt to see if he has any insight.

@monkeyman192 Are you interested in looking at the base moving code in No Man's Connect? It stopped working in the Next update. [Function is in app/app.tsx](https://github.com/jaszhix/NoMansConnect/blob/c051b57632563d505d94c1be6454036e6865cae6/app/app.tsx#L736). I attempted to fix it in August, but it seems there's more to it that I'm missing - couldn't get the base parts to align with the correct up orientation.

**Comment by monkeyman192** [comment]
Gosh, it has been a while since I looked at it. Can you remember what change occurred after next that caused it to break? IIRC it just worked by determining the transformation matrix between the two coordinate systems and applying it right?

**Comment by jaszhix** [comment]
Yeah, as far as I can tell. One difference between how bases work now vs pre-Next builds was before you could have bases start with no base parts, and now it seems you always have at least one because you need to place a base computer. It seems the game is handling the base geometry differently internally in any case.
